### PR TITLE
Made bigquery table schema diffsuppress func handle nested lists gracefully

### PR DESCRIPTION
**Pull request by melinath**
Resolved https://github.com/hashicorp/terraform-provider-google/issues/9230. The comparison function was assuming that if the current object was a list, it was the top-level schema list and could be sorted by field name. However, this logic ended up also getting applied to nested lists, which don't have a field name to sort by (thus causing a panic.)

Specifically, bigquery table schema can specify [policyTags](https://cloud.google.com/bigquery/docs/column-level-security-intro) per-column.

I believe the purpose of sorting by field name is to prevent diffs if fields in the schema are reordered. We can preserve this logic by pre-sorting the schema by field name prior to passing it to the comparison function.

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
bigquery: Fixed `google_bigquery_table.schema` handling of policyTags
```

### CODE REVIEW

File: mmv1/third_party/terraform/resources/resource_bigquery_table.go [hunk 1]
```diff
@@ -147,13 +139,34 @@ func bigQueryTableSchemaDiffSuppress(_, old, new string, _ *schema.ResourceData)
...
+
+	arrayA := a.([]interface{})
+	arrayB, ok := b.([]interface{})
+	if !ok {
+		return false
+	} else if len(arrayA) != len(arrayB) {
+		return false
+	}
+
+	// Sort the first level of the schema by name so reordering the
```
Comment (rileykarson) [inline mmv1/third_party/terraform/resources/resource_bigquery_table.go:156]: A field can have fields nested inside of it [[schema](https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#tablefieldschema)], so we still need to sort within the JSON function still, right? With this change reordering nested `fields` fields won't work as I'm reading it.

Comment (melinath) [reply mmv1/third_party/terraform/resources/resource_bigquery_table.go:156]: Thanks, I didn't know about nested fields! I'll make this support both.

File: mmv1/third_party/terraform/tests/resource_bigquery_table_test.go [hunk 2]
```diff
@@ -11,6 +11,235 @@ import (
...
+					"type": "INTEGER",
+					"description": "otherVal"
+				}
+			]`,
+			ExpectDiffSuppress: false,
+		},
+		"nested field ordering changes": {
+			Old: `[
+				{
+					"name": "someValue",
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tests/resource_bigquery_table_test.go:158]: For the sake of completeness, can we reorder an additional top-level field in this example?

Alternatively, adding a "complete" example that:

* reorders top level fields
* reorders nested fields
* includes a policy_tags value on both a top level and nested field

could be a good idea- one of the problems with the `schema` field is that we haven't captured enough complexity in the tests, historically.

**Review by rileykarson — APPROVED** [review]

File: mmv1/third_party/terraform/tests/resource_bigquery_table_test.go (hunk 2 above)
Comment (melinath) [reply mmv1/third_party/terraform/tests/resource_bigquery_table_test.go:158]: sure, done

**Review by rileykarson — APPROVED** [review]

**Pull request merged**